galley/pkg/config/analysis/analyzers/analyzers_test.go

```diff
@@ -99,6 +99,16 @@ var testGrid = []testCase{
 			{msg.PodMissingProxy, "Pod/default/noninjectedpod"},
 		},
 	},
+	{
+		name: "istioInjectionVersionMismatch",
+		inputFiles: []string{
+			"testdata/injection-with-mismatched-sidecar.yaml",
+		},
+		analyzer: &injection.Analyzer{},
```

Comment:
Ah, sure, I can do that; there's definitely a lot in common between the two that I can extract out.

Should the old file still be called injector.go? They both seem to have a larger scope than 'general validation of the injector'

Reply:
It would probably be good to rename the new file.

I'm also OK with letting the split be a separate PR, so it's easier to track new functionality vs reorganized functionality.

Reply:
I split it out into two separate files - PTAL.

Note that both analyzers don't share code, currently. I considered factoring out code to find all injected namespaces, for example, but I think the existing code is better as it makes the iterating loops over resources very explicit.

Reply:
One use-case for these analyzers is to run in Galley, which can potentially trigger analysis fairly often. We have a debounce mechanism in the snapshotter that provides some rate limiting, but we do want to be careful about how "expensive" these analyzers get, especially when looking at large clusters. So any indexing operation that we can lazily calculate once and share between analyzers, we probably should. 

That said, we don't have a mechanism for such shared indexing yet, because nothing has so far needed it. I'm OK punting on that for now. 

In this particular case with the namespaces, the question of what logic should be shared or not is a bit messy and deserves some TLC to find the right implementation. But I do think at least some of the logic should be shared.